libnmstate/state.py

```diff
@@ -336,10 +338,49 @@ def merge_routes(self, other_state):
                 }
 
         merged_routes = (other_routes | self_routes) - absent_routes
+
+        merged_routes -= self._routes_to_auto_iface(merged_routes)
```

Comment:
I find it suitable to `_is_interface_routable`. Is there a problem to place the logic there?

Reply:
The `_is_interface_routable` only checks on routes copying from current state, it does not check
on routes specified in desired state.

Reply:
Right, but I consider that a good thing: Changing from static to dynamic is about implicitly dropping the existing routes (handling the merging).

Handling the desired state is a validation step, not a merging one. Note that for the desired state we should fail and not silently drop, it is an explicit user request which we do not support at the moment.